metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_audit.py

```diff
@@ -84,7 +84,19 @@ class BigqueryTableIdentifier:
 
     @staticmethod
     def get_table_and_shard(table_name: str) -> Tuple[str, Optional[str]]:
-        match = re.search(
+        """
+        Args:
+            table_name:
+                table name (in form <table-id> or <table-prefix>_<shard>`, optionally prefixed with <project-id>.<dataset-id>)
```

Comment:
why have we changed from get_raw_table() to .table elsewhere in the code, if this one supports the project / dataset prefix?

Reply:
Due to this ->

```
 If table_name is fully qualified, i.e. prefixed with <project-id>.<dataset-id> then the special case of
                dataset as a sharded table, i.e. table-id itself as shard can not be detected.
```
get_table_and_shard("project.dataset.20230112") -> (project.dataset.20230112, None)


get_table_and_shard("20230112") -> (None, 20230112)

Alternatively - we can also fix the regex but it seemed safer to change input instead.

Reply:
Is this a normal practice with sharding? Or have we just seen some customers put a bunch of tables with date names in a dataset and consider them "shards". For example, https://cloud.google.com/bigquery/docs/partitioned-tables#dt_partition_shard says:
> Table sharding is the practice of storing data in multiple tables, using a naming prefix such as [PREFIX]_YYYYMMDD.

Can we be explicit about what cases we identify tables as sharding? It seems like right now, we support:
- `[PREFIX]_YYYYMMDD`
- ~~`[PREFIX]$YYYYMMDD`~~ nvm, seems like `$` is a forbidden character
- `YYYYMMDD`

Reply:
that's right. above two case are what we support with default sharded pattern. the second case (just YYYYMMDD as name) is more of a specific case adopted by very few AFAIK. 

The complexity is that the sharded pattern can be overridden using recipe config  - `sharded_table_pattern` . Would be good to understand if this is really used/required.